browser-test/src/support/applicant_program_list.ts

```diff
@@ -0,0 +1,139 @@
+
+  /**
+   * Get the locator for program card heading in the desired group
+   * @param {ApplicationStatusCardGroupName} applicationStatusCardGroupName to find
+   * @param {String} programName to find
+   * @returns {Locator} Locator for program card in the desired group
+   */
+  getCardHeadingLocator(
+    applicationStatusCardGroupName: ApplicationStatusCardGroupName,
+    programName: string,
+  ): Locator {
+    return this.getCardGroupLocator(applicationStatusCardGroupName).getByRole(
+      'heading',
+      {name: programName},
+    )
+  }
+
+  /**
+   * Clicks the apply button for the specified program within the specified card group
+   *
+   * @async
+   * @param {ApplicationStatusCardGroupName} applicationStatusCardGroupName to find
+   * @param {String} programName to apply to
+   */
+  async clickApplyButton(
+    applicationStatusCardGroupName: ApplicationStatusCardGroupName,
+    programName: string,
+  ): Promise<void> {
+    await this.getCardLocator(applicationStatusCardGroupName, programName)
+      .getByRole('link', {
+        name: this.getCardGroupApplyButtonText(applicationStatusCardGroupName),
+      })
+      .click()
+  }
+
+  /**
+   * @private
+   * Get the label text for the specific card group
+   *
+   * @param {ApplicationStatusCardGroupName} applicationStatusCardGroupName to find
+   * @returns {String} String of the context specific apply button label text
+   */
+  private getCardGroupApplyButtonText(
+    applicationStatusCardGroupName: ApplicationStatusCardGroupName,
+  ): ApplicationStatusApplyButtonText {
+    return applicationStatusCardGroupName ==
+      ApplicationStatusCardGroupName.MyApplications
+      ? ApplicationStatusApplyButtonText.Edit
+      : ApplicationStatusApplyButtonText.ViewAndApply
+  }
+
+  /**
+   * Get the eligibility tag for the desired application status card group name
+   * @param {ApplicationStatusCardGroupName} applicationStatusCardGroupName to find
+   * @returns {Locator} Locator to the eligible tag
+   */
+  getCardEligibleTagLocator(
+    applicationStatusCardGroupName: ApplicationStatusCardGroupName,
+  ): Locator {
+    return this.getCardGroupLocator(applicationStatusCardGroupName).locator(
+      '.cf-eligible-tag',
+    )
+  }
+
+  /**
+   * Get the not eligibility tag for the desired application status card group name
+   * @param {ApplicationStatusCardGroupName} applicationStatusCardGroupName to find
+   * @returns {Locator} Locator to the not eligible tag
+   */
+  getCardNotEligibleTagLocator(
+    applicationStatusCardGroupName: ApplicationStatusCardGroupName,
+  ): Locator {
+    return this.getCardGroupLocator(applicationStatusCardGroupName).locator(
+      '.cf-not-eligible-tag',
+    )
+  }
+}
+
+/**
+ * @readonly
+ * @enum {string}
+ * List of heading names used for different groups of application cards used in this test
+ * suite. This is used in relation to what the user sees on the `/programs` page
+ */
+export enum ApplicationStatusCardGroupName {
```

Comment:
**suggestion (non-blocking):** We (I) might want to add the 'Programs based on your selections' and 'Other programs and services' card groups.  Could we name the enum `CardGroupName`?  Or, if we want to follow the naming in the templates, we could call it `CardSectionName`.

Reply:
Works for me. Much easier to read. Since there is no definition in the designs that I could see on what the collection of cards should be called, I fell back to `CardSectionName` to be consistent with the templates.